docs/index.rst

```diff
@@ -1,49 +1,135 @@
-#######################
-Developer Documentation
-#######################
+####################################
+Open edX Developer Documentation Hub
+####################################
 
-This the landing page for documentation relevant to developers of the Open edX platform.  For other documentation
-resources targeted for edX learners, educators, researchers, and Open edX operators, see `docs.edx.org <docs.edx.org>`_.
+Welcome new and returning Open edX developers! 
+
+This is the central repository for documentation that will help you work with Open edX code. 
+
+
+New to edX? Start here
+======================
+
+This quick overview is intended for developers new to the Open edX platform, junior and senior alike. 
+
+After you've reviewed these topics you'll be ready for more detail on our architecture, or you can get ready for active development with the topics in "Information for all Open edX developers."
+
+* :doc:`new_to_edx/what_you_can_do`
+* :doc:`new_to_edx/mission`
+* :doc:`new_to_edx/what_our_systems_do`
+* :doc:`new_to_edx/data_model`
+* :doc:`new_to_edx/messaging_model`
```

Comment:
These two feel like they should be further down, perhaps in "Working with the core Open edX Platform".  They are more detailed than the other links here, and aren't needed until you are going to write code in the platform.

Reply:
I'm open to this. @robrap ?